### PR TITLE
frame-omni-bencher short checks

### PR DESCRIPTION
**Pull request by mordamax**
- Part of https://github.com/paritytech/ci_cd/issues/1006
- Closes: https://github.com/paritytech/ci_cd/issues/1010
- Related: https://github.com/paritytech/polkadot-sdk/pull/4405

- Possibly affecting how frame-omni-bencher works on different runtimes: https://github.com/paritytech/polkadot-sdk/pull/5083

Currently works in parallel with gitlab short benchmarks. 
Triggered only by adding `GHA-migration` label to assure smooth transition (kind of feature-flag). 
Later when tested on random PRs we'll remove the gitlab and turn on by default these tests

### CODE REVIEW

**Review by alvicsam — APPROVED** [review]

File: .github/workflows/check-frame-omni-bencher.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        run: |
+          RUNTIME_BLOB_NAME=$(echo $PACKAGE_NAME | sed 's/-/_/g').compact.compressed.wasm
+          RUNTIME_BLOB_PATH=./target/release/wbuild/$PACKAGE_NAME/$RUNTIME_BLOB_NAME
+          forklift cargo build --release --locked -p $PACKAGE_NAME -p frame-omni-bencher --features runtime-benchmarks
+          echo "Running short benchmarking for PACKAGE_NAME=$PACKAGE_NAME and RUNTIME_BLOB_PATH=$RUNTIME_BLOB_PATH"
+          ls -lrt $RUNTIME_BLOB_PATH
+          ./target/release/frame-omni-bencher v1 benchmark pallet --runtime $RUNTIME_BLOB_PATH --all --steps 2 --repeat 1
+  confirm-frame-omni-benchers-passed:
+    runs-on: ubuntu-latest
+    name: All migrations passed
```
Comment (ggwpez) [inline .github/workflows/check-frame-omni-bencher.yml:82]:
```suggestion
    name: All benchmarks passed
```

File: .github/workflows/reusable-check-changed-files.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          predicate-quantifier: "every"
+          # current-workflow - check if the current (caller) workflow file is changed
+          # rust - check if any Rust (build-related) file is changed
+          filters: |
+            current-workflow:
+              - '${{ steps.current-file.outputs.current-workflow-file }}'
+            rust:
+              - '**/*'
+              - '!.github/**/*'
```
Comment (ggwpez) [inline .github/workflows/reusable-check-changed-files.yml:57]: Would it make sense to use forklift to decide if something changed?

**Review by ggwpez — APPROVED** [review]

File: .github/workflows/reusable-check-changed-files.yml (hunk 2 above)
Comment (AndWeHaveAPlan) [reply .github/workflows/reusable-check-changed-files.yml:57]: forklift will know about changes only during `cargo build` and this action is for skipping entire workflows if no rust code has changed

Comment (ggwpez) [reply .github/workflows/reusable-check-changed-files.yml:57]: Yea but there could be a `forklift --did-something-change`? Trying to manually re-list all files that could trigger a build seems more fragile. But maybe its not the objective of forklift to offer this...

**Review by EgorPopelyaev — APPROVED** [review]

**Comment by alvicsam** [comment]
cc @bkontur

**Comment by bkontur** [comment]
> cc @bkontur

@alvicsam @mordamax guys, iiuc, this is merged, but it is not enabled yet, right? So, how can I enable this new CIs?

My plan is to finish this week setting up testnet runtimes to work with frame-omni-bencher (not all the runtime work): https://github.com/paritytech/polkadot-sdk/pull/5083 + https://github.com/paritytech/polkadot-sdk/pull/5083

**Comment by alvicsam** [comment]
@bkontur yep, that's correct, this is not enabled yet. If you want to add it to the current CI you need to change gitlab ci files.

**Comment by mordamax** [comment]
@bkontur i think if you could fix 2-3 failing frame omni bencher checks with merging those fixes, may be you could just remove the gitlab part of it and remove [this condition](https://github.com/paritytech/polkadot-sdk/blob/master/.github/workflows/check-frame-omni-bencher.yml#L21) - then it will be fully migrated to GHA and frame omni bencher by default

**Comment by mordamax** [comment]
@bkontur please don't add `GHA-migration` label unless you are going to test/fix those, as there's still limited capacity for new runners :)

**Comment by bkontur** [comment]
> @bkontur please don't add `GHA-migration` label unless you are going to test/fix those, as there's still limited capacity for new runners :)

@mordamax don't worry, thanks for that conditions (exactly, what I was looking for: https://github.com/paritytech/polkadot-sdk/pull/4405/commits/a47aff48ab72abb213c4e1ee84d3da44849829a3), I will continue [here](https://github.com/paritytech/polkadot-sdk/pull/4405) with fixing and when everything works we will remove gitlab CI pipelines

**Pull request merged**